src/base/model/smodel.ts

```diff
@@ -110,7 +110,6 @@ export class SParentElement extends SModelElement {
             throw new Error(`No such child ${child.id}`);
         }
         children.splice(i, 1);
-        delete (child as {parent: SParentElement}).parent;
```

Comment:
Instead of just omitting these lines, the `parent` property should at least be set to `undefined`.

Reply:
Basically, this is more or less the same as "delete". Because "parent” is defined as: `readonly parent: SParentElement;` which cannot be undefined this is not allowed. As mentioned here: https://github.com/eclipse/sprotty/issues/218 it would be required to define it as `parent?: SParentElement;` which effects a lot of code that needs to add "undefined" checks.